src/common/functions.js

```diff
@@ -0,0 +1,117 @@
+export const s4 = () => {
+  return Math.floor((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .substring(1);
+};
```

Comment:
What is this function doing? returning a random string/substring? 

Add comments for what it does

Reply:
Also, you check if this id already exists, not sure if you are doing that